routes/donations.js

```diff
@@ -0,0 +1,177 @@
+const express = require('express');
+const { keysToCamel } = require('../common/utils');
+const { db } = require('../server/db');
+
+const router = express.Router();
+
+// get all donation rows
+router.get('/', async (req, res) => {
+  try {
+    const allDonations = await db.query(`SELECT * FROM donations;`);
+    res.status(200).json(allDonations);
+  } catch (err) {
+    res.status(500).send(err.message);
+  }
+});
+
+// get specific donation
+router.get('/:donationId', async (req, res) => {
+  try {
+    const { donationId } = req.params;
+    const donation = await db.query(`SELECT * from donations WHERE id = $(donationId);`, {
+      donationId,
+    });
+    res.status(200).json(keysToCamel(donation));
+  } catch (err) {
+    res.status(500).send(err.message);
+  }
+});
+
+// create new donation
+router.post('/', async (req, res) => {
+  try {
+    const {
+      routeId,
+      orderNum,
+      status,
+      addressStreet,
+      addressUnit,
+      addressCity,
+      addressZip,
+      firstName,
+      lastName,
+      email,
+      phoneNum,
+      notes,
+      date,
+    } = req.body;
+    const donation = await db.query(
+      `INSERT INTO donations (
+        ${routeId ? 'route_id, ' : ''}
+        ${orderNum ? 'order_num, ' : ''}
+        address_street,
+        ${addressUnit ? 'address_unit, ' : ''}
+        address_city, address_zip, first_name,
+        last_name, email, phone_num,
+        ${notes ? 'notes, ' : ''}
+        ${date ? 'date, ' : ''}
+        status
+        )
+      VALUES (
+        ${routeId ? '$(routeId), ' : ''}
+        ${orderNum ? '$(orderNum), ' : ''}
+        $(addressStreet),
+        ${addressUnit ? '$(addressUnit), ' : ''}
+        $(addressCity), $(addressZip), $(firstName),
+        $(lastName), $(email), $(phoneNum),
+        ${notes ? '$(notes), ' : ''}
+        ${date ? '$(date), ' : ''}
+        $(status)
+      )
+      RETURNING *;`,
+      {
+        routeId,
+        orderNum,
+        status,
+        addressStreet,
+        addressUnit,
+        addressCity,
+        addressZip,
+        firstName,
+        lastName,
+        email,
+        phoneNum,
+        notes,
+        date,
+      },
+    );
+    res.status(200).send(keysToCamel(donation[0]));
```

Comment:
we recently changed the backend route standard to always return an array, even if it's just one element, so you can just return `donation` by itself